compiler/hie-core/src/Development/IDE/Core/Shake.hs

```diff
@@ -352,13 +352,13 @@ shakeRun IdeState{shakeExtras=ShakeExtras{..}, ..} acts = modifyVar shakeAbort $
     bar <- newBarrier
     start <- offsetTime
     thread <- forkFinally (shakeRunDatabaseProfile shakeProfileDir shakeDb acts) $ \res -> do
-        signalBarrier bar res
         runTime <- start
         let res' = case res of
                 Left e -> "exception: " <> displayException e
                 Right _ -> "completed"
         logDebug logger $ T.pack $
             "Finishing shakeRun (took " ++ showDuration runTime ++ ", " ++ res' ++ ")"
+        signalBarrier bar res
```

Comment:
I’ve rearranged this since I got confused when I saw the logging output in line 351 before this one and thought our synchronization was broken.